Source/DafnyRuntime/DafnyRuntime.py

```diff
@@ -72,16 +72,37 @@ def c_label(name: str = None):
         if g.target != name:
             raise g
 
+class CodePoint(str):
+
+    escapes = {
+      '\n' : "\\n",
+      '\r' : "\\r",
+      '\t' : "\\t",
+      '\0' : "\\0",
+      '\'' : "\\'",
+      '\"' : "\\\"",
+      '\\' : "\\\\",
+    }
+
+    def __escaped__(self):
+        return self.escapes.get(self, self)
+
+    def __dafnystr__(self):
+        return f"'{self.__escaped__()}'"
+
 class Seq(tuple):
     def __init__(self, __iterable = None, isStr = False):
         if __iterable is None:
             __iterable = []
-        self.isStr = isStr \
-                     or isinstance(__iterable, str) \
-                     or (isinstance(__iterable, Seq) and __iterable.isStr) \
-                     or (not isinstance(__iterable, GeneratorType)
-                         and all(isinstance(e, str) and len(e) == 1 for e in __iterable)
-                         and len(__iterable) > 0)
+        if isStr is None:
```

Comment:
Why would `isStr` ever be `None`. Do we not expect it to be passed in as either `False` or `True`? Or is the idea to support this third value in order to give different behavior?

Reply:
The latter, and I added a big chunky docstring to explain what's going on here.